### PR TITLE
Optimize master query taskInstance's new state from cache.

### PR DESCRIPTION
**Pull request by tlhhup**
## What is the purpose of the pull request
In my understanding, the worker updates the status of the task through the message after the master has distributed the task, but the new task status in the master is directly obtained from the database without the use of TaskInstanceCacheManager. So I optimized the code, If there is any problem, please correct it.😊

1. after master submit the task to worker and db, cache it.
2. when receive the task state update response from worker, update the cache in addition to synchronizing to the db.
3. query new state from cache, reduce the pressure on the db.

This change added tests and can be verified as follows:
  - *Manually verified the change by testing locally.*

### CODE REVIEW

**Comment by break60** [comment]
Pull the latest code

**Comment by break60** [comment]
Please add ut

**Comment by qiaozhanwei** [comment]
great 

1，pid meaningless, no need to add，can remove
2，taskInstance cache no problem，but there are many places involving master code，please commit dev and test

**Comment by tlhhup** [comment]
thanks,
1. save process id  in zk is to solve this [issue](https://github.com/apache/incubator-dolphinscheduler/issues/2735). I solve it on another PR, but checkout new branch with this commit, so it can be see here.
2. i will add the UT. BTW, could you give me some advices?

**Comment by qiaozhanwei** [comment]
Unit Test not running . please retry commit . Thx

**Comment by tlhhup** [comment]
> Unit Test not running . please retry commit . Thx

I check the log detail, it is blocked by some connect problem, such as hadoop or http. Do need  modify some configurations? or Can you give some advice?

**Comment by yangyichao-mango** [comment]
Hi,
You can change one blank line to rerun the ut.

File: dolphinscheduler-server/src/test/java/org/apache/dolphinscheduler/server/master/processor/TaskAckProcessorTest.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+import java.net.InetSocketAddress;
+
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.when;
+import static org.powermock.api.mockito.PowerMockito.mock;
+
+
+/**
+ * @author 离歌笑
```
Comment (lgcareer) [inline dolphinscheduler-server/src/test/java/org/apache/dolphinscheduler/server/master/processor/TaskAckProcessorTest.java:44]: please remove the author information,thx

**Comment by CalvinKirs** [comment]
Hi, please follow the reviewer's suggestion to change it. If there is no update after 7 days, the PR will be closed, and you can reopen it when your problem is resolved.

**Pull request closed without merging**